elasticdl/python/tests/pserver_servicer_test.py

```diff
@@ -88,10 +92,54 @@ def testServicer(self):
         res = self._stub.pull_embedding_vector(req)
         self.assertEqual(res, elasticdl_pb2.Tensor())
 
-        req = elasticdl_pb2.Model()
-        res = self._stub.push_model(req)
-        self.assertEqual(res, empty_pb2.Empty())
-
         req = elasticdl_pb2.PushGradientRequest()
         res = self._stub.push_gradient(req)
         self.assertEqual(res, elasticdl_pb2.PushGradientResponse())
+
+    def testPushModel(self):
+        self.create_default_server_and_stub()
+        param0 = {
+            "v0": np.random.rand(3, 2).astype(np.float32),
+            "v1": np.random.rand(10, 32).astype(np.float32),
+        }
+        param1 = {
+            "v0": np.ones([3, 2], dtype=np.float32),
+            "v1": np.ones([10, 32], dtype=np.float32),
+        }
+        embedding_info = elasticdl_pb2.EmbeddingTableInfo()
+        embedding_info.name = "layer0"
+        embedding_info.dim = 32
+        embedding_info.initializer = "normal"
+
+        models = [param0, param1]
+
+        for idx, model in enumerate(models):
+            req = elasticdl_pb2.Model()
+            req.version = idx + 1
+            for name in model:
+                req.param.append(ndarray_to_tensor(model[name], name))
+            req.embedding_table_info.append(embedding_info)
+            res = self._stub.push_model(req)
+            self.assertEqual(res, empty_pb2.Empty())
+            # self._parameters is initialized with the first push_model call
+            # and the second push_model has no effect
+            self.assertEqual(self._parameters.version, 1)
+            for name in param0:
+                np.allclose(
+                    param0[name],
+                    self._parameters.non_embedding_params[name].numpy(),
+                )
```

Comment:
Is this intentional or forget to call `self.assertTrue`?

Reply:
Fixed.